hvac/api/auth_methods/kerberos.py

```diff
@@ -0,0 +1,485 @@
+import fnmatch
+import logging
+import os
+import string
+from os.path import isfile
+from random import random
+from urllib.parse import urlparse
+
+import gssapi
+import gssapi.raw as gssapi_raw
+from gssapi.raw.misc import GSSError
+from requests_gssapi import HTTPSPNEGOAuth
+from requests_gssapi.exceptions import SPNEGOExchangeError
+
+from hvac import utils
+from hvac.api.auth_methods.ldap import Ldap as LdapAuth
+from hvac.exceptions import Unauthorized
+
+DEFAULT_MOUNT_POINT = "kerberos"
+HOST_TEMPLATE = '_HOST'
+ENV_KRB5_CONF = 'KRB5_CONFIG' # (optional) environment variable tells us the path to your krb5.conf file.
+ENV_KRB5_CACHE = 'KRB5CCNAME'  # (optional) environment variable tells us the directory of your kerberos credential cache
+SPNEGO_OID = '1.3.6.1.5.5.2'
+
+
+class KerberosError(Exception):
+    """
+    Please keep this exception so users can differentiate between kerberos issues and general api errors.
+    """
+    pass
+
+
+class Kerberos(LdapAuth):  # TODO for discussion do you want ldap subclassed or to be verbose and a bit redundant ?
+    """Kerberos Auth Method (API).
+
+    Reference: https://www.vaultproject.io/api/auth/kerberos/index.html
+    """
+    def _log(self):
+        return logging.getLogger(self.__class__.)
+
+    def configure(self, **kwargs):
+        super().configure(**kwargs)
+        # TODO - for discussion, what should be configurable
+        # krb5_conf, etc?
+
+    def read_configuration(self, mount_point=DEFAULT_MOUNT_POINT):
+        """
+        Retrieve the KERBEROS configuration for the auth method.
+
+        Supported methods:
+            GET: /auth/{mount_point}/config. Produces: 200 application/json
+
+        :param mount_point: The "path" the method/backend was mounted on.
+        :type mount_point: str | unicode
+        :return: The JSON response of the read_configuration request.
+        :rtype: dict
+        """
+        api_path = utils.format_url(
+            "/v1/auth/{mount_point}/config", mount_point=mount_point
+        )
+        return self._adapter.get(
+            url=api_path,
+        )
+
+    def create_or_update_group(
+        self, name, policies=None, mount_point=DEFAULT_MOUNT_POINT
+    ):
+        """
+        Create or update LDAP group policies.
+
+        Supported methods:
+            POST: /auth/{mount_point}/groups/{name}. Produces: 204 (empty body)
+
+
+        :param name: The name of the LDAP group
+        :type name: str | unicode
+        :param policies: List of policies associated with the group. This parameter is transformed to a comma-delimited
+            string before being passed to Vault.
+        :type policies: list
+        :param mount_point: The "path" the method/backend was mounted on.
+        :type mount_point: str | unicode
+        :return: The response of the create_or_update_group request.
+        :rtype: requests.Response
+        """
+        return super().create_or_update_group(name, policies, mount_point)
```

Comment:
Though we will probably want to avoid subclassing anyway, please consistently use named arguments instead of positional for calling. There is a chance we will look to make many arguments named-only in the future.